### ISSUE TITLE
Not working with DSM 7.1.1-42962 Update 1

### ISSUE THREAD

**Issue by Lodifice** [runfalk/synology-wireguard#168]
Hi, I have a DS 220j and with the recent upgrade to 7.1.1-42962 Update 1, WireGuard stopped working. I rebuilt the package and installed it again, but without success.

The intended (and previously working) use case is a VPN client inside a home network (behind NAT) that connects to a server running WireGuard. When I tcpdump on the server, I can see packets coming from the client and responses going out, but according to the WireGuard logs, the handshake never completes. It's strange that the port number of the client's endpoint entry is reported as 0 (which is not a valid UDP port). When I tcpdump on the client, I can see ICMP messages that the packets to the invalid UDP port 0 are dropped (these are the server responses, apparently) and the RX counter of `wg show` stays at zero.

Moreover, once setup the wg0 interface cannot be brought down, as `ip link del dev wg0` segfaults. Then the entire system slowly becomes unresponsive and a forced shutdown is the only option.

Has anyone experienced a similar issue? Or can someone give some pointers on where to dig further?

**Comment by sergey-brutsky** [comment]
Hello, I have DS 220j with upgrade `DSM 7.1-42661 Update 4`

Compiled manually with params `docker run --rm --privileged --env PACKAGE_ARCH=rtd1296 --env DSM_VER=7.0 -v $(pwd)/artifacts:/result_spk synobuild`

Installed as mentioned in the guide, everything works fine for me.

**Comment by zeldovich** [comment]
@Lodifice I also ran into this problem with 7.1.1-42962, on a DS418.  As far as I can tell, there's something buggy with the resulting `wg` kernel module.  I'm not quite sure what's going on, but perhaps Synology applied some changes to the kernel for 7.1.1-42962, compared to the kernel from 7.1, which affected the networking functions used by `wg.ko`.  The kernel version number is the same (which is why modules built using kernel headers from 7.1 seemingly load on 7.1.1 to begin with), but that doesn't say much.  Synology hasn't posted a build toolchain for 7.1.1 specifically (I opened an issue about it, https://github.com/SynologyOpenSource/pkgscripts-ng/issues/42).

**Comment by Lodifice** [comment]
@sergey-brutsky Thanks for you're answer, but you're running an older version of DSM than I. With your version, I didn't have problems either. Also check your `DSM_VER`, for some reason it's 7.1.

@zeldovich it's good to know I'm not the only one with that problem, thanks for opening the issue.

**Comment by stpkys** [comment]
Same problem with `DS418j` and `DSM 7.1.1-42962`.

**Comment by Single568** [comment]
Hi, guys same problem on DS420j version DSM 7.1.1-42962 Update 1. Before update work without problems. When i tried some wg command via SSH it ended with hard restart, system totaly not responding :-(. So i uninstal wireguard and compile with [this modification](https://github.com/runfalk/synology-wireguard/issues/154#issuecomment-1169393940) and with this version in docker command: DSM_VER=7.1 But same story, again system not responding :-(.
Any help, please?

**Comment by stpkys** [comment]
Seems like wireguard kernel module does not work for 7.1.1. No idea about the root cause, but segfaults and system crashes are too scary to continue investigation. Fortunately, NAS is restarting by itself and getting back to life after some time. 

As a proof of concept I was able to connect via user-space client https://github.com/WireGuard/wireguard-go 
I'm going to migrate to this setup and hopefully it's not going to be significantly slower.

**Comment by ruimarinho** [comment]
Working well for me on 918+ with both 7.1.1 Update 1 and 7.1.1 Update 2.

**Comment by Nightreaver** [comment]
> Working well for me on 918+ with both 7.1.1 Update 1 and 7.1.1 Update 2.

How did you compile your version then?

**Comment by spielmap** [comment]
> > Working well for me on 918+ with both 7.1.1 Update 1 and 7.1.1 Update 2.
> 
> How did you compile your version then?

We used a workflow for Github actions: https://github.com/vegardit/synology-wireguard/blob/master/.github/workflows/build.yml
Images can be found here: https://github.com/vegardit/synology-wireguard/releases/tag/latest-DSM7.1

**Comment by Nightreaver** [comment]
> > > Working well for me on 918+ with both 7.1.1 Update 1 and 7.1.1 Update 2.
> > 
> > 
> > How did you compile your version then?
> 
> We used a workflow for Github actions: https://github.com/vegardit/synology-wireguard/blob/master/.github/workflows/build.yml Images can be found here: https://github.com/vegardit/synology-wireguard/releases/tag/latest-DSM7.1

Amazing! let me try that

**Comment by spielmap** [comment]
> > > > Working well for me on 918+ with both 7.1.1 Update 1 and 7.1.1 Update 2.
> 
> > > 
> 
> > > 
> 
> > > How did you compile your version then?
> 
> > 
> 
> > We used a workflow for Github actions: https://github.com/vegardit/synology-wireguard/blob/master/.github/workflows/build.yml Images can be found here: https://github.com/vegardit/synology-wireguard/releases/tag/latest-DSM7.1
> 
> 
> 
> Amazing! let me try that
> 
> 
> 
> Did you try on DSM 7.1.1-42962 Update 2 ?

Yes, the modules loads successfully. Haven't tried to bring up a tunnel yet. Did it work for you?

**Comment by Nightreaver** [comment]
It seems to work well, at least for accessing my local network with it. 
I need to tweak my iptables/routing, but tunneling works fine.

thank a lot for the effort

**Comment by spielmap** [comment]
New URL: https://github.com/vegardit/synology-wireguard/releases
We now build releases for 6.2 and 7.1 for all supported platforms.

**Comment by Single568** [comment]
> New URL: https://github.com/vegardit/synology-wireguard/releases We now build releases for 6.2 and 7.1 for all supported platforms.

Same story for me `DS420j` version `DSM 7.1.1-42962 Update 1`

**Comment by Geier97332** [comment]
It doesn't work for me too. DS118

**Comment by mycroft2k** [comment]
On DS220j with DSM 7.1.1-42962 Update 2 its not working, when WireGuard starts, NAS stops responding, automatic reboot..
DS220j only wokring with DSM 7.0.1-42218 Update 5 and WireGuard works.

DS220+ and DS920+ (DSM 7.1.1-42962 Update 2)  it works perfectly

I think all NAS higher than DSM 7.0.1-42218 Update 5 with Realtek RTD1296 SoC have the problem

**Comment by AlexeyGG** [comment]
Just wondering if anyone managed to solve this issue?
I've got the same Segmentation fault error with wireguard after upgrading to latest  "DSM 7.1.1-42962 Update 6'.
I stuck with the issue as I can't rollback this update.

**Comment by stpkys** [comment]
@AlexeyGG,  I couldn't find a solution to this issue, but I managed to find a workaround by using the [userspace version of WireGuard](https://github.com/WireGuard/wireguard-go)

Here are the steps to follow:

1. Compile the ARM version of WireGuard-Go from the repository https://github.com/WireGuard/wireguard-go.
1. Copy the `wireguard-go` binary to `/var/packages/WireGuard/target/wireguard` (same folder with `wg-quick` file)
1. Remove or rename the wireguard.ko file in the same folder.
1. Verify that the `tun` module is loaded by running `lsmod | grep tun`. If it's not loaded, you can load it using `insmod tun`.
1. Now you can try to use WireGuard as usual.

This workaround is a bit clumsy, but I can try to publish a fork in the next couple of days to simplify the process.

**Comment by o850cHQk** [comment]
> 

That would be great if you could. Going to give this a go since I've had to put something in front of one of my offsite backups in the interim hoping for some kind of fix.
